### PR TITLE
Add dartdoc for RenderBox

### PR DESCRIPTION
**Pull request by abarth**

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: sky/packages/sky/lib/src/rendering/box.dart [hunk 1]
```diff
@@ -259,17 +291,19 @@ abstract class RenderBox extends RenderObject {
...
-  // this node during that parent's performLayout() or paint().
+
+  /// Returns the distance from the y-coordinate of the position of the box to
+  /// the y-coordinate of the first given baseline in the box's contents.
+  ///
+  /// This is used by certain layout models to align adjacent boxes on a common
+  /// baseline, regardless of padding, font size differences, etc. If there is
+  /// no baseline, and the 'onlyReal' argument was not set to true, then it
+  /// returns the distance from the y-coordinate of the position of the box to
+  /// the y-coordinate of the bottom of the box, i.e., the height of the box.
```
Comment (Hixie) [inline sky/packages/sky/lib/src/rendering/box.dart]: Say what happens if onlyReal _is_ true.

File: sky/packages/sky/lib/src/rendering/box.dart [hunk 2]
```diff
@@ -259,17 +291,19 @@ abstract class RenderBox extends RenderObject {
...
+  ///
+  /// Only call this function after layout has been performed. You are only
```
Comment (Hixie) [inline sky/packages/sky/lib/src/rendering/box.dart]: "Only call this function on a box after its layout() method has been called." maybe better.

File: sky/packages/sky/lib/src/rendering/box.dart [hunk 3]
```diff
@@ -363,6 +403,15 @@ abstract class RenderBox extends RenderObject {
     assert(sizedByParent);
   }
 
+  /// Determines the set of render objects located at the given position
+  ///
+  /// Returns true if the given point is contained in this render object of one
```
Comment (Hixie) [inline sky/packages/sky/lib/src/rendering/box.dart]: of -> or

File: sky/packages/sky/lib/src/rendering/box.dart [hunk 4]
```diff
@@ -441,7 +518,16 @@ abstract class RenderBox extends RenderObject {
...
   }
 
+  /// Returns a rectangle that contains all the pixels painted by this box
+  ///
+  /// The paint bounds can be larger or smaller than [size], which is the amount
+  /// of space this box takes up during layout. For example, if this box casts a
+  /// shadow, that shadow might extend beyond the space allocated to this box
+  /// during layout.
+  ///
+  /// The returned paint bounds are in the local coordinate system of this box.
```
Comment (Hixie) [inline sky/packages/sky/lib/src/rendering/box.dart]: explain why this exists

File: sky/packages/sky/lib/src/rendering/box.dart [hunk 5]
```diff
@@ -483,13 +569,17 @@ abstract class RenderBox extends RenderObject {
   String debugDescribeSettings(String prefix) => '${super.debugDescribeSettings(prefix)}${prefix}size: ${size}\n';
 }
 
-
-// HELPER METHODS FOR RENDERBOX CONTAINERS
+/// A mixin that provides useful default behaviors for boxes with children
```
Comment (Hixie) [inline sky/packages/sky/lib/src/rendering/box.dart]: "... with children managed by the [ContainerRenderObjectMixin] mixin"

**Comment by Hixie** [comment]
somehow I thought the RenderBox protocol was more complicated than that. Huh.

LGTM.

File: sky/packages/sky/lib/src/rendering/box.dart (hunk 1 above)
Comment (abarth) [inline sky/packages/sky/lib/src/rendering/box.dart]: Done.

File: sky/packages/sky/lib/src/rendering/box.dart (hunk 2 above)
Comment (abarth) [inline sky/packages/sky/lib/src/rendering/box.dart]: Fixed.

File: sky/packages/sky/lib/src/rendering/box.dart (hunk 3 above)
Comment (abarth) [inline sky/packages/sky/lib/src/rendering/box.dart]: Fixed

File: sky/packages/sky/lib/src/rendering/box.dart (hunk 4 above)
Comment (abarth) [inline sky/packages/sky/lib/src/rendering/box.dart]: Fixed.

File: sky/packages/sky/lib/src/rendering/box.dart (hunk 5 above)
Comment (abarth) [inline sky/packages/sky/lib/src/rendering/box.dart]: Done.

**Pull request merged**